internal/internal_workflow_client.go

```diff
@@ -1835,13 +1835,23 @@ func (w *workflowClientInterceptor) UpdateWorkflow(
 		},
 	})
 	if err != nil {
-		return nil, err
+		var de *serviceerror.DeadlineExceeded
+		if !errors.As(err, &de) {
+			return nil, err
+		}
```

Comment:
I do not think we can assume a failed gRPC request, regardless of reason, means that it is an acceptable update. Can we discuss the server-side reasons why this would ever occur? The server should never fail on this call if it should be successful. If that means it needs to return within the long-poll (e.g. 55s before the 60s timeout or whatever) before it times out, so be it.